### DISCUSSION TITLE
Is it possible to customize openai's api address?

### DISCUSSION THREAD

**Discussion by moeik** [vanna-ai/vanna#257] [Q&A] [answered]
I would like to customize the api address of openai because I can't use the services of openai directly in my location. I'm going to use nginx to proxy openai's api address, and then change the request address to my proxy api address

**Comment by zainhoda** [comment, marked as answer]
Yes, you can pass in your own OpenAI client object where you specify the endpoint.

```python
class MyVanna(ChromaDB_VectorStore, OpenAI_Chat):
    def __init__(self, config=None):
        ChromaDB_VectorStore.__init__(self, config=config)
        OpenAI_Chat.__init__(self, client=YOUR_OPENAI_CLIENT, config=config) # Make sure to put your AzureOpenAI client here

vn = MyVanna(config={'model': 'gpt-4-...'})
```

Alternatively, you can set your `OPENAI_BASE_URL` environment variable

https://github.com/openai/openai-python/blob/main/src/openai/_client.py#L306
